mud-contracts/world/src/modules/smart-gate/systems/SmartGateSystem.sol

```diff
@@ -121,7 +121,7 @@ contract SmartGateSystem is EveSystem, AccessModified {
    * @param destinationGateId is the smartObjectId of the destination gate
    */
   function linkSmartGates(uint256 sourceGateId, uint256 destinationGateId) public onlyAdminOrObjectOwner(sourceGateId) {
-    if (isGateLinked(sourceGateId, destinationGateId)) {
+    if (isAnyGateLinked(sourceGateId, destinationGateId)) {
```

Comment:
the function should be more like`isGateDestinationLinked`

```sol
function isGateDestinationLinked(uint256 sourceGateId) public view returns (bool) {
  SmartGateLinkTableData memory smartGateLinkTableData = SmartGateLinkTable.get(sourceGateId);

  return smartGateLinkTableData.isLinked;
}
```

because we allow a gate to have multiple sources right? But only one destination

Reply:
same as above

Reply:
If either the source or the destination gate is linked you shouldn't be able to link it together. So it is correct in the new PR. Gates can't be connected to multiple sources. You can only link A <-> B. We might change that at some point in time.

Reply:
if we don't allow A -> B, B -> C chain link then it's ok!